src/line.c

```diff
@@ -69,14 +69,13 @@ get_line_info(const char *prefix, enum line_type type)
 	struct line_info *info;
 	struct line_rule *rule;
 
-	/* XXX: Map special highligh types to avoid special settings. */
-	if (type == LINE_DIFF_ADD_HIGHLIGHT)
-		type = LINE_DIFF_ADD;
-	else if (type == LINE_DIFF_DEL_HIGHLIGHT)
-		type = LINE_DIFF_DEL;
-
 	assert(type < line_rules);
 	rule = &line_rule[type];
+
+	/* special highlight types are not prefix-dependent */
+	if ((type == LINE_DIFF_ADD_HIGHLIGHT) || (type == LINE_DIFF_DEL_HIGHLIGHT))
```

Comment:
I think it would make sense to allow `stage.diff-add-highlight` to use a different color in the stage view, but I don't feel strongly about that.